commit c6b8191374d9ad064eb96423400a6314c2d0102e
Author: Kinglong Mee <kinglongmee@gmail.com>
Date:   Tue Jun 30 14:12:38 2015 -0400

    blkmapd: Fix infinite loop when reading serial
    
    If (dev_id->ids & 0xf) < current_id, must updates pos when continue.
    Otherwise an infinite loop.
    
    No other places use the pos value, just move to the top of while.
    
    Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/blkmapd/device-inq.c b/utils/blkmapd/device-inq.c
index c5bf71f..6b56b67 100644
--- a/utils/blkmapd/device-inq.c
+++ b/utils/blkmapd/device-inq.c
@@ -196,6 +196,8 @@ struct bl_serial *bldev_read_serial(int fd, const char *filename)
 
 	while (pos < (len - devid_len)) {
 		dev_id = (struct bl_dev_id *)&(dev_root->data[pos]);
+		pos += (dev_id->len + devid_len);
+
 		if ((dev_id->ids & 0xf) < current_id)
 			continue;
 		switch (dev_id->ids & 0xf) {
@@ -226,7 +228,6 @@ struct bl_serial *bldev_read_serial(int fd, const char *filename)
 		}
 		if (current_id == 3)
 			break;
-		pos += (dev_id->len + devid_len);
 	}
  out:
 	if (!serial_out)
